tests/checks/test_cross_join_check.py

```diff
@@ -305,3 +305,51 @@ def test_with_cross_join_with_limit(self):
         with self.patch_schema(schema):
             stmt = sqlparse.parse(sql)[0]
             assert self.has_cross_join(stmt)
+
+    def test_with_no_cross_join_with_inner_order(self):
+        tbl_a, tbl_b = Mock(), Mock()
+        tbl_a.primary_key.columns.keys.return_value = ['arrowid']
+        tbl_b.primary_key.columns.keys.return_value = ['bookid']
+        schema = {
+            'arrow': tbl_a,
+            'book': tbl_b
+        }
+        sql = (
+            "SELECT * "
+            "FROM ("
+            "    SELECT arrowid AS objectid, "
+            "    'arrow' as string "
+            "    FROM arrow "
+            "    WHERE 1=1 "
+            "    ORDER BY arrow.arrowid"
+            "  UNION ALL"
+            "    SELECT bookid AS objectid, "
+            "    'book' as string "
+            "    FROM book"
+            ") as anon1 "
+            "ORDER BY anon1.objectid DESC, anon1.string ASC"
+        )
+        with self.patch_schema(schema):
+            stmt = sqlparse.parse(sql)[0]
+            assert False == self.has_cross_join(stmt)
+
+    def test_with_cross_join_with_order(self):
```

Comment:
Since this is simpler test, I would move it before the `test_with_no_cross_join_with_inner_order`. Simple to complex rule. 🙂